src/common/components/UniversalDialog/types.ts

```diff
@@ -0,0 +1,17 @@
+import { ButtonProps, DialogProps } from '@mui/material'
+
+import { SetStateAction, Dispatch } from 'react'
+
+export interface UniversalDialogProps extends DialogProps {
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
+  heading?: string
+  content?: string
+  primaryButtonText?: string
+  secondaryButtonText?: string
+  primaryButtonProps?: Omit<ButtonProps, 'children' | 'onClick'>
+  secondaryButtonProps?: Omit<ButtonProps, 'children' | 'onClick'>
```

Comment:
Why not just accept `JSX.Element`?

Reply:
To limit customizability. If we accept JSX.Element, we can do something like 
```tsx
secondaryButtonProps: {
 children: () => (<>
                    <Typography variant='subtitle1'>brah brah brah</Typography>
                    <Typography variant='subtitle2'>brah brah brah</Typography>
                 </>)
}
``` 
That's not good for our Design System.

If you want to add an icon, you can use `startIcon` or `endIcon` of `ButtonProps`